openvdb/tools/MeshToVolume.h

```diff
@@ -1123,13 +1123,16 @@ class SeedFillExteriorSign
         for (size_t n = range.begin(), N = range.end(); n < N; ++n) {
             if (mChangedNodeMask[n]) {
                 //seedFill(*mNodes[n]);
-                mChangedNodeMask[n] = scanFill(*mNodes[n]);
+                //Whether or not scanFill changes any values, the node
+                //has still changed since the last attempt to propagate
+                //to adjacent nodes. The mask is therefore left unchanged.
+                scanFill(*mNodes[n]);
```

Comment:
Hey @tristanbarback, I've finally had some time to look at this, thanks for your patience. I believe your changes are correct but I have a minor suggestion - though disclaimer that I haven't looked closely at this code until recently. 

As you've pointed out, the problem is isolated in the two operators `SeedFillExteriorSign` and `SeedPoints`. `SeedFillExteriorSign` updates individual leaf nodes with sign information based purely on the current leaf nodes data i.e it doesn't query neighbouring leaf nodes. `SeedPoints` then attempts to propagate changes that have occurred locally within each leaf to the faces of neighbouring leaf nodes. The issue is that only leaf nodes which have had their masks changed by `SeedFillExteriorSign` are updated in `SeedPoints`, where in fact a leaf node which is unchanged by `SeedFillExteriorSign` may still need to propagate sign information into itself from updated neighbours.

However I think we can retain some of the previous optimisations. At first glance this particular change within `SeedFillExteriorSign` seemed unnecessary to me - simply forcing `SeedPoints` to run over every leaf node should be enough to ensure the correct sign propagation is enforced between the necessary neighbours(?). However, if on the very first run (of the loop inside of `traceExteriorBoundaries`) a particular leaf node isn't changed at all, then `mChangedNodeMask` will be set to false and this will be picked up by the leaf neighbour queries in `SeedPoints` - meaning that although `SeedPoints` is run for every leaf node, sign propagation never occurs. I believe this is only ever an issue on the *first* execution of this loop and we can safely set `mChangedNodeMask` on subsequent iterations. The changes I'm proposing are as follows:

1) On the first run of `SeedFillExteriorSign`, make sure every leaf node is processed and leave `mChangedNodeMask` untouched (all true). This ensures that even if a node isn't changed on the first run, nodes check their neighbouring face signs.

```
    void operator()(const tbb::blocked_range<size_t>& range) const {
        if (!mChangedNodeMask) {
            for (size_t n = range.begin(), N = range.end(); n < N; ++n) {
                scanFill(*mNodes[n]);
            }
        }
        else {
            for (size_t n = range.begin(), N = range.end(); n < N; ++n) {
                if (mChangedNodeMask[n]) {
                    //seedFill(*mNodes[n]);
                    mChangedNodeMask[n] = scanFill(*mNodes[n]);
                }
            }
        }
    }
```

2) Update `SeedPoints` to always run over every node and query neighbour offsets. Your current changes already do this.
3) For subsequent runs of `SeedFillExteriorSign`, we can now update and track node changes with `mChangedNodeMask`.

```
    bool* changedMasks = nullptr;
    do {
        tbb::parallel_for(nodeRange, mesh_to_volume_internal::SeedFillExteriorSign<FloatTreeT>(
            nodeConnectivity.nodes(), changedMasks));

        tbb::parallel_for(nodeRange, mesh_to_volume_internal::SeedPoints<FloatTreeT>(
            nodeConnectivity, changedNodeMaskA.get(), changedNodeMaskB.get(),
            changedVoxelMask.get()));

        changedNodeMaskA.swap(changedNodeMaskB);
        changedMasks = changedNodeMaskA.get();
        ...
    }
```

Let me know what you think.